mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -572,13 +593,29 @@ public void write(DataOutput out) throws IOException {
       byte[] data = SerializationUtil.serializeToBytes(this.task);
       out.writeInt(data.length);
       out.write(data);
+
+      byte[] ioData = SerializationUtil.serializeToBytes(io);
```

Comment:
@rdsr, I just noticed this. Do we need custom `Serializable` implementations? If all of the fields are `Serializable` and this is `Serializable` then the default implementation should just work.

Reply:
If we want to verify this, we can add a test for serializability, too.

Reply:
sounds good. I'll double check this.

Reply:
@rdblue, can you elaborate more on what is the default implementation here?  I don't think there's a default implementation for Serializable in MR. There's a default impl. for Writables

Reply:
Java serialization will work without implementing `write` and `readFields` as long as all of the fields in an object are `Serializable`. The only reason to implement this is to wrap a non-serializable object or to customize in some way how serialization happens. Because all 3 fields that need to be serialized are already `Serializable` (`FileScanTask`, `FileIO`, and `EncryptionManager`) I don't think we need these methods.

Reply:
To follow up on this, I don't think that we need to remove it in this PR. But we should be able to remove these in a follow-up.